modules/core/src/effects/effect-manager.js

```diff
@@ -0,0 +1,22 @@
+export default class EffectManager {
+  constructor() {
+    this._effects = [];
+  }
+
+  addEffect(effect) {
```

Comment:
Fine, but consider that effects will be specified in a declarative way, rather than imperative:

```
deck.setProps({
  effects: [new Effect..., new Effect]
});
```

We may not have much need of `addEffect` and `removeEffect` but rather a `setEffects`.

Maybe add an API Audit note in a tracker task.

Reply:
Another thing: Ultimately we might want to do diffing on the effects, like we do on the layers. This was actually a major reason I started to break out the lifecycle handling from the `Layer` class. See code in `lib/lifecycle`. 

I don't expect that understanding that code is very easy, so not expecting this to be done now. Just raising awareness that prop diffing should be possible to add later, to improve perf when we have many effects.

Reply:
agree